addon/components/frost-icon.js

```diff
@@ -1,22 +1,54 @@
-import Ember from 'ember'
-import { PropTypes } from 'ember-prop-types'
 import _ from 'lodash'
+import Ember from 'ember'
+const {Component, deprecate} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import PropTypeMixin, {PropTypes} from 'ember-prop-types'
+
+export default Component.extend(PropTypeMixin, {
```

Comment:
I believe I had run into issues with `getDefaultProps()` not working properly without the mixin. Maybe it was unrelated but I'm not sure.